### PR TITLE
Introduce `JDBC2` version store type, deprecate `JDBC` version store type

### PR DESCRIPTION
**Pull request by snazy**
The current `JDBC` version store type uses a lot of columns in the `objs` table, which cause quite some overhead in storage. The new `JDBC2` version store type works like `JDBC`, but uses a single column for all object values and also serializes object-ids as binary values.

### CODE REVIEW

**Comment by snazy** [comment]
The difference is not that much, but still a bit:

Running

```bash
java -jar tools/content-generator/build/libs/nessie-content-generator-0.95.1-SNAPSHOT.jar generate -b 10 -n 20000 -t 500
```

yields via 

```sql
SELECT
    table_schema || '.' || table_name AS table_full_name,
    pg_size_pretty(pg_total_relation_size('"' || table_schema || '"."' || table_name || '"')) AS size
FROM test.information_schema.tables
WHERE table_schema = 'public'
ORDER BY
    pg_total_relation_size('"' || table_schema || '"."' || table_name || '"') DESC;
```

``` 
public.objs,484 MB
public.objs2,469 MB
public.refs,1792 kB
public.refs2,72 kB
```
(objs + refs = JDBC, objs2 + refs2 = JDBC2)

File: versioned/storage/jdbc2/src/main/java/org/projectnessie/versioned/storage/jdbc2/DatabaseSpecifics.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+import java.sql.Types;
+import java.util.EnumMap;
+import java.util.Locale;
+import java.util.Map;
+import javax.sql.DataSource;
+
+public final class DatabaseSpecifics {
+  private DatabaseSpecifics() {}
+
+  // Use 'ucs_basic' collation for PostgreSQL, otherwise multiple spaces would be collapses and
```
[the remark this answers is not in the document]
Comment (snazy) [reply versioned/storage/jdbc2/src/main/java/org/projectnessie/versioned/storage/jdbc2/DatabaseSpecifics.java:31]: Eh - that typo is years old :p

File: versioned/storage/jdbc/src/intTest/java/org/projectnessie/versioned/storage/jdbc/ITCockroachDBBackendFactory.java [hunk 2]
```diff
@@ -15,9 +15,11 @@
  */
 package org.projectnessie.versioned.storage.jdbc;
 
+import org.junit.jupiter.api.Disabled;
 import org.projectnessie.versioned.storage.jdbctests.AbstractJdbcBackendTestFactory;
 import org.projectnessie.versioned.storage.jdbctests.CockroachBackendTestFactory;
 
+@Disabled("Disabled in favor of jdbc2")
```
[the remark this answers is not in the document]
Comment (snazy) [reply versioned/storage/jdbc/src/intTest/java/org/projectnessie/versioned/storage/jdbc/ITCockroachDBBackendFactory.java:22]: Just to save some time in CI.

File: versioned/storage/jdbc2/src/main/java/org/projectnessie/versioned/storage/jdbc2/Jdbc2BackendBaseConfig.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  /**
+   * The JDBC catalog name.
+   *
+   * @deprecated This setting has never worked as expected and is now ineffective. The catalog must
+   *     be specified directly in the JDBC URL using the option {@code
+   *     quarkus.datasource.*.jdbc.url}.
+   */
+  @Deprecated(forRemoval = true)
+  Optional<String> catalog();
```
[the remark this answers is not in the document]
Comment (snazy) [reply versioned/storage/jdbc2/src/main/java/org/projectnessie/versioned/storage/jdbc2/Jdbc2BackendBaseConfig.java:32]: removed

File: versioned/storage/jdbc/src/intTest/java/org/projectnessie/versioned/storage/jdbc/ITCockroachDBBackendFactory.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dimas-b) [reply versioned/storage/jdbc/src/intTest/java/org/projectnessie/versioned/storage/jdbc/ITCockroachDBBackendFactory.java:22]: fair enough

File: versioned/storage/jdbc2/src/main/java/org/projectnessie/versioned/storage/jdbc2/Jdbc2BackendBaseConfig.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (dimas-b) [reply versioned/storage/jdbc2/src/main/java/org/projectnessie/versioned/storage/jdbc2/Jdbc2BackendBaseConfig.java]: I just realized that `JdbcBackendBaseConfig.catalog()` and `.schema()` are no longer settable, so can probably remove them too.

**Review by dimas-b — COMMENTED** [review]
LGTM. Do we want to remove `JdbcBackendBaseConfig.catalog()` and `.schema()`?

File: versioned/storage/jdbc2/src/main/java/org/projectnessie/versioned/storage/jdbc2/Jdbc2BackendBaseConfig.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (dimas-b) [reply versioned/storage/jdbc2/src/main/java/org/projectnessie/versioned/storage/jdbc2/Jdbc2BackendBaseConfig.java]: Actually, if users have (old) config setting for those properties, they will get config validation errors now... Do we want to mention that in the CHANGELOG?

[the remark this answers is not in the document]
Comment (snazy) [reply versioned/storage/jdbc2/src/main/java/org/projectnessie/versioned/storage/jdbc2/Jdbc2BackendBaseConfig.java]: removed

**Review by dimas-b — APPROVED** [review]

**Pull request merged**